src/Routes/HomePage/HomePage.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react';
+import {axiosInstance} from "../../component/axios/axios";
+
+export function HomePage(){
+    const [isAuthenticated, setIsAuthenticated] = useState(false);
+    const [products, setProducts] = useState([]);
+  
+    useEffect(() => {
+      const authStatus = checkAuth(); 
```

Comment:
Remember to create a function outside useEffect so your code is easier to read. This may not seem important now since your logic is short but it could grow in the future. 
I would add `const authStatus = checkAuth();` and `setIsAuthenticated(authStatus);` inside checkAuth function.

Reply:
I applied the reccomended changes in the new commit. Thanks!

Reply:
Left one more comment about this logic